### PR TITLE
Nominated Trader initial implementation

### PR DESCRIPTION
**Pull request by gangov**

### CODE REVIEW

File: contracts/trader/src/tests/setup.rs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        .deploy(factory_wasm)
+}
+
+pub fn deploy_and_init_lp_client(
+    env: &Env,
+    admin: Address,
+    token_a: Address,
+    token_b: Address,
+    token_a_amount: i128,
+    token_b_amount: i128,
```
Comment (ueco-jb) [inline contracts/trader/src/tests/setup.rs:44]: Dude. 
```suggestion
    token_a: Address,
    token_a_amount: i128,
    token_b: Address,
    token_b_amount: i128,
```

**Review by ueco-jb — COMMENTED** [review]
Plus add at least one fully fleged case, where there are 3 tokens, 2 liquidity pools (with fees as it will be). Different ratios in the pools before trade. Trade both of them to $PHO and include some math comments to prove the results.

**Review by ueco-jb — COMMENTED** [review]
Cool, now I see that I didn't publish these comments for a week already.

File: contracts/trader/src/contract.rs [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+            .publish(("Trader: Initialize", "PHO token: "), output_token);
+    }
+
+    fn trade_token(
+        env: Env,
+        sender: Address,
+        token_to_swap: Address,
+        liquidity_pool: Address,
+        amount: Option<u64>,
+        max_spread: Option<u64>,
```
[the remark this answers is not in the document]
Comment (gangov) [reply contracts/trader/src/contract.rs]: decided to go with `Option<i64>` because this is the type we use in pool

**Review by ueco-jb — COMMENTED** [review]
LGTM

**Pull request merged**